pkg/ddl/job_submitter.go

```diff
@@ -324,11 +324,18 @@ func (s *JobSubmitter) addBatchDDLJobs2Table(jobWs []*JobWrapper) error {
 		if job.CDCWriteSource == 0 && bdrRole != string(ast.BDRRoleNone) {
 			if job.Type == model.ActionMultiSchemaChange && job.MultiSchemaInfo != nil {
 				for _, subJob := range job.MultiSchemaInfo.SubJobs {
-					if DeniedByBDR(ast.BDRRole(bdrRole), subJob.Type, job) {
+					var args model.JobArgs
+					if subJob.Type == model.ActionAddIndex {
```

Comment:
we need check `action == model.ActionAddPrimaryKey` too, see DeniedByBDR

Reply:
maybe save `JobArgs` directly inside `SubJob` with `json:"-"`, it's harder to add a wrapper like struct for it now, for that we can refactor later.